app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,103 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication::AuthnAzure
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+      dependencies: {
+        role_class:                 ::Role,
+        resource_class:             ::Resource,
+        application_identity_class: ApplicationIdentity,
+        logger:                     Rails.logger
+      },
+      inputs:       %i(account service_id username xms_mirid_token_field oid_token_field)
+    ) do
+
+      def call
+        validate_xms_mirid_format
+        token_identity_from_claims
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
```

Comment:
Let's call the method in a name that will remove the comment above it. `call` methods should be readable without comments.

Reply:
changing to `compare_token_id_with_annotations`

Reply:
identity*